src/common/config.cpp

```diff
@@ -56,4 +115,21 @@ void Config::loadModelParameters(const void* ptr) {
   override(config);
 }
 
+void Config::override(const YAML::Node& params) {
+  for(auto& it : params) {
+    config_[it.first.as<std::string>()] = it.second;
+  }
+}
+
+void Config::log() {
+  YAML::Emitter out;
+  cli::OutputYaml(config_, out);
```

Comment:
Why do we need this complicated interface? Why is there no single function in YAML that does this and just returns a string? I suggest to add such a function. The Emitter protocol seems an irrelevant concern at this level of code.

Reply:
Copied over from previous code, just moved from *.h to *.cpp.

Reply:
Added TODO